api/client/client.go

```diff
@@ -828,23 +828,21 @@ func (c *Client) GetKubeServices(ctx context.Context) ([]types.Server, error) {
 
 // GetApplicationServers returns all registered application servers.
 func (c *Client) GetApplicationServers(ctx context.Context, namespace string) ([]types.AppServer, error) {
-	resp, err := c.grpc.GetApplicationServers(ctx, &proto.GetApplicationServersRequest{
-		Namespace: namespace,
-	}, c.callOpts...)
+	resources, err := c.ListAllResources(ctx, namespace, proto.ResourceType_RESOURCE_TYPE_APPLICATION_SERVER)
 	if err != nil {
-		if trace.IsNotImplemented(trail.FromGRPC(err)) {
-			servers, err := c.getApplicationServersFallback(ctx, namespace)
-			if err != nil {
-				return nil, trace.Wrap(err)
-			}
-			return servers, nil
-		}
-		return nil, trail.FromGRPC(err)
+		return nil, trace.Wrap(err)
 	}
-	var servers []types.AppServer
-	for _, server := range resp.GetServers() {
-		servers = append(servers, server)
+
+	servers := make([]types.AppServer, len(resources))
+	for i, resource := range resources {
+		appServer, ok := resource.(types.AppServer)
+		if !ok {
+			return nil, trace.Errorf("invalid resource type %T", resource)
```

Comment:
We don't usually use `trace.Errorf` directly. Use `trace.BadParameter` by default or a more appropriate trace error. Also, make the error message more useful.
```suggestion
			return nil, trace.BadParameter("expected AppServer resource, got %T", resource)
```